### PR TITLE
fixed roundtrip test for stage file item

### PR DESCRIPTION
**Pull request by jo-pol**
fixed error handling, size zero as default in EasyFileMetadata

### CODE REVIEW

**Comment by jo-pol** [comment]
ingest fails, while trying to add relations: can't find the key in `pidDictionary(relation.objectSDO)`

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala [hunk 1]
```diff
@@ -15,17 +15,18 @@
...
 
 object EasyFileMetadata {
-  def apply(s: FileItemSettings): Elem =
+  def apply(s: FileItemSettings): Try[String] = Try {
       <fimd:file-item-md xmlns:fimd="http://easy.dans.knaw.nl/easy/file-item-md/" version="0.1" >
         <name>{s.pathInDataset.get.getName}</name>
         <path>{s.pathInDataset.get}</path>
         <mimeType>{s.format.getOrElse("application/octet-stream")}</mimeType>
-        <size>{s.size.get}</size>
+        <size>{s.size.getOrElse(0)}</size>
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala:26]: Not sure about the "or else". No size specified should be an error, I think.

**Comment by jo-pol** [comment]
what we are now producing looks inconsistent with https://github.com/DANS-KNAW/easy-ingest/#digital-object-configuration-file 

```
"namespace":"easy-file",
"datastreams":[{
  "dsLocation":"http://dummy.dans.knaw.nl/ook/maar/wat.mpg",
  "dsID":"EASY_FILE",
  "controlGroup":"R",
  "mimeType":"video/mpeg"
},{
  "contentFile":"EASY_FILE_METADATA",
  "dsID":"EASY_FILE_METADATA",
  "controlGroup":"X",
  "mimeType":"text/xml"
}],
```

**Comment by jo-pol** [comment]
aha, `cfg.json` is missing `info:fedora/` in case of `isMemberOf`

```
"relations":[{
  "predicate":"http://dans.knaw.nl/ontologies/relations#isMemberOf",
  "object":"easy-dataset:1"
},{
  "predicate":"http://dans.knaw.nl/ontologies/relations#isSubordinateTo",
  "object":"info:fedora/easy-dataset:1"
}
```

**Comment by jo-pol** [comment]
but the webui shows weird duplicated folders after the session below:
duplicated

```
rm -rf out/
easy-stage-file-item  -i easy-dataset:2 --format video/mpeg -p original/tif/myvideo1.mpg -d http://dummy.dans.knaw.nl/ook/maar/wat.mpg out
easy-stage-file-item  -i easy-dataset:2 --format video/mpeg -p original/new/myvideo2.mpg -d http://dummy.dans.knaw.nl/ook/maar/wat.mpg out
easy-stage-file-item  -i easy-dataset:2 --format video/mpeg -p new/myvideo3.mpg -d http://dummy.dans.knaw.nl/ook/maar/wat.mpg out
easy-stage-file-item  -i easy-dataset:2 --format video/mpeg -p myvideo4.mpg -d http://dummy.dans.knaw.nl/ook/maar/wat.mpg out
easy-ingest out/easy-dataset_2/
easy-update-fs-rdb easy-dataset:2
easy-update-solr-index -i easy-dataset:2
sudo service tomcat6 restart
```

**Comment by jo-pol** [comment]
@janvanmansum 

The query in getExistingAncestor does not yet test for the datasetID and optional trailing slash, as discussed.

In the round trip test commands above a pause is required before restarting tomcat and your dataset may not have the tif-folder as mine does. I also noticed it is better to start with a fresh dataset than reusing one were the round trip test stumbled half way.

**Comment by jo-pol** [comment]
@DANS-KNAW/easy 

All problems addressed above are fixed

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala (hunk 1 above)
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala:26]: Size zero doesn't seem to cause a problem in the round trip test. To show the real value would require a http-header request with the specified url, if that provides the length.

Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala:26]: I don't understand your comment about an HTTP-header. The size is provided to `easy-stage-file-item` on the command line or in the csv file. `easy-stage-dataset` gets the size from the file itself.

Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFileMetadata.scala:26]: Sorry, I should have checked that. Now I understand your remark about default zero that should be an error: the option should be made mandatory, when no csv is specified.

**Comment by jo-pol** [comment]
squashed in #60

**Pull request closed without merging**